server/assets-v1/templates/src/pages/user_portal/password_reset/reset-password-page.html

```diff
@@ -1,178 +1,205 @@
-    const newPassword = ref("");
-    const repeatedNewPassword = ref("");
+    <script>
+        const { createApp, computed, ref } = Vue;
 
-    const messageToSign = "Sign-in with Layer8";
+        const username = ref("");
+        const mnemonicSentence = ref("");
+        const newPassword = ref("");
+        const repeatedNewPassword = ref("");
 
-    const resetPassword = async () => {
-        if (username.value === "" || mnemonicSentence.value === ""
+        const messageToSign = "Sign-in with Layer8";
+
+        const resetPassword = async () => {
+            if (username.value === "" || mnemonicSentence.value === ""
                 || newPassword.value === "" || repeatedNewPassword.value === "") {
-            alert("All fields are mandatory");
-            return;
-        }
-        if (newPassword.value !== repeatedNewPassword.value) {
-            alert("Repeated password does not match");
-            return;
-        }
+                alert("All fields are mandatory");
+                return;
+            }
+            if (newPassword.value !== repeatedNewPassword.value) {
+                alert("Repeated password does not match");
+                return;
+            }
 
-        let currMnemonic = mnemonicSentence.value.trim();
+            let currMnemonic = mnemonicSentence.value.trim();
 
-        if (!mnemonic.isValid(currMnemonic)) {
-            alert("The provided mnemonic is invalid");
-            return;
-        }
+            if (!mnemonic.isValid(currMnemonic)) {
+                alert("The provided mnemonic is invalid");
+                return;
+            }
 
-        const keyPair = mnemonic.getPrivateAndPublicKeys(currMnemonic);
-        const signature = mnemonic.sign(keyPair.privateKey, messageToSign);
+            const keyPair = mnemonic.getPrivateAndPublicKeys(currMnemonic);
+            const signature = mnemonic.sign(keyPair.privateKey, messageToSign);
 
-        try {
-            const response = await window.fetch(
-                    "[[ .ProxyURL ]]/api/v1/reset-password",
+            try {
+                const responseOne = await window.fetch(
+                    "[[ .ProxyURL ]]/api/v2/reset-password-precheck",
                     {
                         method: "POST",
                         headers: {
                             "Content-Type": "application/json",
                         },
                         body: JSON.stringify({
                             username: username.value,
-                            signature: Array.from(signature),
-                            new_password: newPassword.value
                         }),
                     },
-            );
+                );
 
-            const result = await response.json();
+                const resetPasswordPrecheckResponseBody = await responseOne.json();
+                if (responseOne.status !== 200) {
+                    alert("Something wen wrong!");
```

Comment:
I think it's better to add an error message into the alert displayed to the user. Something like that:
`alert("Error: " + result.message);`